### DISCUSSION TITLE
Problems with flatsurface in version 0.13

### DISCUSSION THREAD

**Discussion by EdwinSoft** [jimy-byerley/pymadcad#46] [Q&A] [answered]
Dear All,

Congratulations for this excellent project and I have a question, with version 0.12 of pymadcad the following code worked:

`
from madcad import *

ext = wire(Circle((O, -Y), 15, resolution=('div', 6)))
cir = wire(Circle((O, Y), 10))
surf = flatsurface([ext, cir])
show([surf])
`
![img](https://user-images.githubusercontent.com/5746631/191124792-c4897eb7-eed1-4366-8cc0-dab73897c9d3.png)
Trying the same code with version 0.13, I get an error:

> Traceback (most recent call last):
  File "/home/edwin/PycharmProjects/pymv13/main.py", line 5, in <module>
    surf = flatsurface([ext, cir])
  File "/home/edwin/PycharmProjects/pymv13/venv/lib/python3.10/site-packages/madcad/generation.py", line 401, in flatsurface
    m = triangulation.triangulation(web(outline), normal)
  File "/home/edwin/PycharmProjects/pymv13/venv/lib/python3.10/site-packages/madcad/mesh/conversions.py", line 63, in web
    pool.mergeclose()
  File "/home/edwin/PycharmProjects/pymv13/venv/lib/python3.10/site-packages/madcad/mesh/container.py", line 73, in mergeclose
    self.mergepoints(merges)
  File "/home/edwin/PycharmProjects/pymv13/venv/lib/python3.10/site-packages/madcad/mesh/web.py", line 85, in mergepoints
    del self.edges[j:]
NotImplementedError: Subscript deletion not supported by arrex.list.typedlist

Process finished with exit code 1

did anything change in version 0.13? or how should I modify the code to make it work?

Ubuntu 22.04
Python 3.10

**Comment by jimy-byerley** [comment, marked as answer]
Hello @EdwinSoft !
I apologize for this bug, as the fault is all mine.

Your code is good, but the changes I released in v0.13 are relying on a feature that was missing in `arrex` in the last (currently available) release. I will release it soon.
In the mean time your can either
- downgrade to  `madcad` v0.12
- install the developement version of `arrex` from source

**Reply by jimy-byerley** [reply]
I opened #47 for this

**Comment by EdwinSoft** [comment]
Thanks for the reply Jimy. Meanwhile I will continue working with v0.12

**Reply by jimy-byerley** [reply]
This is released now. You should be able to upgrade pymadcad to v0.13.1 that fixes the dependency issue, and this will automatically upgrade arrex to v0.5 that contains the missing features :)
```bash
pip install -U pymadcad
```

**Reply by EdwinSoft** [reply]
Thanks. Now the program works fine with the update
